Source/UrlRequest_Android.cpp

```diff
@@ -91,8 +91,32 @@ namespace UrlLib
                         URL url{m_appPathOrUrl.data()};
 
                         URLConnection connection{url.OpenConnection()};
-                        connection.Connect();
 
+                        // set request headers
+                        for (auto request : m_requestHeaders)
+                        {
+                            std::string key = request.first;
+                            std::string value = request.second;
```

Comment:
These are unnecessary copies.
```suggestion
                            const std::string& key = request.first;
                            const std::string& value = request.second;
```